src/pyimcom/diagnostics/context_figure.py

```diff
@@ -0,0 +1,58 @@
+"""
+Context manager for writing report figures.
+"""
+
+
+class ReportFigContext:
+    """
+    This is a context manager for report figures in PyIMCOM.
+
+    Parameters
+    ----------
+    mpl : class
+        The matplotlib module.
+    plt : class
+        The pyplot submodule.
+
+    Attributes
+    ----------
+    mpl : class
+        The matplotlib module.
+    plt : class
+        The pyplot submodule.
+    env_backend : str
+        The matplotlib backend in the environment.
+    usetex : bool or None
+        Enviroment self.usetex setting.
+
+    Methods
+    -------
+    __init__
+        Constructor.
+    __enter__
+        Set up local configuration.
+    __exit__
+        Restore old settings.
+
+    """
+
+    def __init__(self, mpl, plt):
+        self.mpl = mpl
+        self.plt = plt
+
+    def __enter__(self):
+        self.env_backend = self.mpl.get_backend()
+        self.usetex = self.plt.rcParams["text.usetex"] if hasattr(self.plt.rcParams, "text.usetex") else None
```

Comment:
Cleaner to write
```
self.usetex = self.plt.rcParams.get("text.usetex", None)
```
where `None` is the fallback value if `text.usetex` is not in the dict

Reply:
This is the one change I would actually suggest be made